### DISCUSSION TITLE
Allow any push to private repository to trigger a deployment

### DISCUSSION THREAD

**Discussion by devotoare** [vercel/community#6769] [Help] [answered]
### Summary

I asked the AI bot about why pushes from other people aren't trigger deployments in my repo and got this answer:

> If the repository is private, commits will only be deployed if the commit author also has access to the respective project on Vercel. This is determined by comparing the Login Connections defined in the Personal Account with the commit author. If the commit author is not the owner of the destination Personal Account, the Deployment will be prevented.

My question is how do I bypass this check? I have users that commit code that I do not want to have any access to Vercel.

### Steps to Reproduce

1. Create private repo.
2. Link to Vercel.
3. Have someone else push to repo.

**Comment by amyegan** [comment, marked as answer]
The AI response was correct. Only commits from a member of the team will trigger a new deployment from a private repo 

[vercel.com/docs/deployments/git#deploying-private-git-repositories](https://vercel.com/docs/deployments/git#deploying-private-git-repositories)

**Reply by brunobely** [reply]
Is this a Vercel-imposed limitation? Why?

**Reply by amyegan** [reply]
It's an additional security measure to make sure only individuals who are explicitly included by a Vercel team owner can make changes to a project

**Reply by tiagoefreitas** [reply]
This is a business model decision, not a security decision, if it was security it would be up to the admin to decide.
Masquerading makes you look bad, and devs are not stupid. 
Would be much better for everyone if you said it outright. This is what happens when a startup starts being partially run by the marketing or sales department.

**Reply by amyegan** [reply]
Hey @tiagoefreitas. Can you explain more about the setup you want to use?

It sounds like you're using a private repo since the restriction does not apply to public repos. You have folks pushing code to that private repo. And you want their commits to create new deployments even if they aren't part of the team and won't be able to view build output or deployment URLs after pushing the commit. Am I understanding your situation?

**Reply by devotoare** [reply]
@amyegan Yes, that is a very common workflow and was the basis of my original question. Though I accepted this was a business model decision and moved on.

[a comment by an automated account is not shown]
